aQute.libg/test/aQute/lib/io/IOTest.java

```diff
@@ -54,4 +55,74 @@ public void testDestDirIsChildOfSource() throws Exception {
 		} catch (IllegalArgumentException e) {
 		}
 	}
+
+	public void testIfCreateSymlinkOrCopyFileDependingOnOS() throws Exception {
+		File link = new File("generated/test/target.dat");
+
+		IO.delete(link);
+
+		assertFalse(link.exists() || Files.isSymbolicLink(link.toPath()));
+
+		link.getParentFile().mkdirs();
+
+		File source = new File("testresources/zipped.dat");
+
+		assertTrue(source.exists());
+
+		assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+		if (IO.isWindows()) {
+			assertFalse(Files.isSymbolicLink(link.toPath()));
+		} else {
+			assertTrue(Files.isSymbolicLink(link.toPath()));
+		}
+	}
+
+	public void testOnlyCopyIfReallyNeededOnWindows() throws Exception {
+		if (IO.isWindows()) {
+			File link = new File("generated/test/target.dat");
+
+			IO.delete(link);
+
+			assertFalse(link.exists() || Files.isSymbolicLink(link.toPath()));
+
+			link.getParentFile().mkdirs();
+
+			File source = new File("testresources/zipped.dat");
+			assertTrue(source.exists());
+
+			assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+			assertEquals(link.lastModified(), source.lastModified());
+			assertEquals(link.length(), source.length());
+
+			assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+			assertEquals(link.lastModified(), source.lastModified());
+			assertEquals(link.length(), source.length());
+		}
+	}
+
+	public void testCreateSymlinkOrCopyWillDeleteOriginalLink() throws Exception {
+		File originalSource = new File("testresources/unzipped.dat");
+		File link = new File("generated/test/originalLink");
+
+		link.delete();
+
+		assertFalse(Files.isSymbolicLink(link.toPath()));
+
+		Files.createSymbolicLink(link.toPath(), originalSource.toPath());
```

Comment:
Is there a risk this `Files.createSymbolicLink` call will fail on some users' systems? It seems to work on the appveyor machines. But some other person building on Windows may not run as admin. I am thinking this call should be

```java
assertTrue(IO.createSymbolicLinkOrCopy(link, originalSource));
```

You should probably also `assertTrue` on the next call to `IO.createSymbolicLinkOrCopy` too.